### PR TITLE
advancements

### PR DESCRIPTION
**Pull request by aaryswastaken**

### CODE REVIEW

**Comment by aaryswastaken** [comment]
The key retrieval logic has been implemented but not tested by any way

File: ether/proto0001.go [hunk 1]
```diff
@@ -74,22 +83,49 @@ func (c *Connection) serve0001() {
...
+			conn := manager.FetchUserById(buff[2:])
+
+			if conn == nil {
+				respBuff := []byte{0xca}
+				respBuff = append(respBuff[:], buff)
+
+				_, err := c.bind.Write(respBuff)
+
+				if err != nil {
+					c.log.Warn("user could not be found")
```
Comment (Portevent) [inline ether/proto0001.go]: Isn't it checking if the response to the client went wrong ? This log probably need to be displayed anyway if we enter `conn == nil`

File: ether/proto0001.go [hunk 2]
```diff
@@ -74,22 +83,49 @@ func (c *Connection) serve0001() {
...
+				continue
+			}
+
+			keyBuff = make([]byte, 2)
+ 			binary.LittleEndian.PutUint16(keyBuff, conn.keyLength) // append the key length as uint16
+			keyBuff = append(keyBuff, conn.key) // append the key 
+
+
+			respBuff := []byte{0xa0, 0xba}
+			respBuff = append(respBuff[:], response[:]) // prepend the pck id
```
Comment (Portevent) [inline ether/proto0001.go]: `respBuff = append(respBuff[:], keyBuff[:])`

**Review by Portevent — CHANGES_REQUESTED** [review]
Two typos, seems alrigth otherwise

**Review by Portevent — APPROVED** [review]

**Pull request merged**